SonarQube.TeamBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -63,53 +67,71 @@ public AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSetting
                 throw new ArgumentNullException("projectKey");
             }
 
-            AnalyzerSettings analyzerSettings = null;
-            if (IsCSharpPluginInstalled(server))
-            {
-                this.sqServer = server;
-                this.sqSettings = settings;
-                this.sqProjectKey = projectKey;
-                this.sqProjectBranch = projectBranch;
+            this.sqServer = server;
+            this.sqSettings = settings;
+            this.sqProjectKey = projectKey;
+            this.sqProjectBranch = projectBranch;
+
+            IList<AnalyzerSettings> analyzersSettings = new List<AnalyzerSettings>();
+            IEnumerable<string> installedPlugins = server.GetInstalledPlugins();
 
-                RoslynExportProfile profile = TryGetRoslynConfigForProject();
+            if (installedPlugins.Contains(CSharpPluginKey))
+            {
+                RoslynExportProfile profile = TryGetRoslynConfigForProject(CSharpLanguage);
                 if (profile != null)
                 {
-                    analyzerSettings = ProcessProfile(profile);
+                    analyzersSettings.Add(ProcessExportedProfile(profile, CSharpLanguage));
                 }
             }
-            else
+            if (installedPlugins.Contains(VBNetPluginKey))
+            {
+                RoslynExportProfile profile = TryGetRoslynConfigForProject(VBNetLanguage);
+                if (profile != null)
+                {
+                    analyzersSettings.Add(ProcessExportedProfile(profile, VBNetLanguage));
+                }
+            }
+
+            if (!analyzersSettings.Any())
             {
                 logger.LogDebug(Resources.RAP_CSharpPluginNotInstalled);
             }
 
-            return analyzerSettings ?? new AnalyzerSettings(); // return emtpy settings rather than null
+            return analyzersSettings;
+        }
+
+        public static string GetRoslynFormatName(string language)
+        {
+            return RoslynFormatNamePrefix + language;
+        }
+
+        public static string GetRoslynRulesetFileName(string language)
+        {
+            return RoslynRulesetFileNamePrefix + language + ".ruleset";
```

Comment:
by switching to formats rather than prefixes, this simply becomes a call to `string.Format()`